app/controllers/admin/pages_controller.rb

```diff
@@ -39,5 +46,21 @@ def page_params
                           social_image]
       params.require(:page).permit(allowed_params)
     end
+
+    def prepopulate_new_form slug
+      # we don't want to pass the html as a param, hence we render it manually with if statement
+        if slug == "code-of-conduct"
+          html = view_context.render partial: "pages/coc_text", locals: { community_name: view_context.community_name, community_qualified_name: view_context.community_qualified_name, email_link: view_context.email_link }
```

Comment:
We pre-populate the new Page record with some information to enhance the user experience for the admin